lib/opal/compiler.rb

```diff
@@ -184,8 +185,16 @@ def parse
 
       begin
         sexp, comments, tokens = @parser.tokenize(@buffer)
+      rescue ::Opal::Error => error
+        backtrace = error.backtrace
+        if error.respond_to? :location
+          line = error.location.line
+          backtrace << "#{file}:#{line}:in #{error.location.expression.source_line}"
```

Comment:
It looks procedural. Maybe we could override a `backtrace` method on our error classes to include this custom line by default? Or it can be another method that does `backtrace + custom_line`

Reply:
I guess the first option doesn't work well - https://bugs.ruby-lang.org/issues/14693 😄